packages/backend/src/tokens/request.ts

```diff
@@ -69,13 +72,64 @@ function isRequestEligibleForRefresh(
   );
 }
 
-export async function authenticateRequest(
+function checkTokenTypeMismatch(
+  parsedTokenType: MachineTokenType,
+  acceptsToken: NonNullable<AuthenticateRequestOptions['acceptsToken']>,
+  authenticateContext: AuthenticateContext,
+): UnauthenticatedState<MachineTokenType> | null {
+  const mismatch = !isTokenTypeAccepted(parsedTokenType, acceptsToken);
+  if (mismatch) {
+    return signedOut({
+      tokenType: parsedTokenType,
+      authenticateContext,
+      reason: AuthErrorReason.TokenTypeMismatch,
+    });
+  }
+  return null;
+}
+
+export interface AuthenticateRequest {
+  /**
+   * @example
+   * clerkClient.authenticateRequest(request, { acceptsToken: ['session_token', 'api_key'] });
+   */
+  <T extends readonly TokenType[]>(
+    request: Request,
+    options: AuthenticateRequestOptions & { acceptsToken: T },
+  ): Promise<RequestState<T[number]>>;
+
+  /**
+   * @example
+   * clerkClient.authenticateRequest(request, { acceptsToken: 'session_token' });
+   */
+  <T extends TokenType>(
+    request: Request,
+    options: AuthenticateRequestOptions & { acceptsToken: T },
+  ): Promise<RequestState<T>>;
+
+  /**
+   * @example
+   * clerkClient.authenticateRequest(request, { acceptsToken: 'any' });
+   */
+  (request: Request, options: AuthenticateRequestOptions & { acceptsToken: 'any' }): Promise<RequestState<TokenType>>;
+
+  /**
+   * @example
+   * clerkClient.authenticateRequest(request);
+   */
+  (request: Request, options?: AuthenticateRequestOptions): Promise<RequestState<SessionTokenType>>;
+}
+
+export const authenticateRequest: AuthenticateRequest = (async (
   request: Request,
   options: AuthenticateRequestOptions,
-): Promise<RequestState> {
+): Promise<RequestState<TokenType>> => {
   const authenticateContext = await createAuthenticateContext(createClerkRequest(request), options);
   assertValidSecretKey(authenticateContext.secretKey);
 
+  // Default tokenType is session_token for backwards compatibility.
```

Comment:
So do we want/need to add a TODO to change this behavior in the future?

Reply:
I don't think there are plants to change the behavior in the future yet. But I will update as soon as I get more info!